app/uk/gov/hmrc/eoricommoncomponent/frontend/services/UserGroupIdSubscriptionStatusCheckService.scala

```diff
@@ -39,23 +39,21 @@ class UserGroupIdSubscriptionStatusCheckService @Inject() (
     save4Later.fetchCacheIds(groupId)
       .flatMap {
         case Some(cacheIds) =>
-          val sameUser    = cacheIds.internalId == internalId
           val sameService = cacheIds.serviceCode.contains(service.code)
 
           subscriptionStatusService
             .getStatus(idType, cacheIds.safeId.id)(hc, service, request)
-            .flatMap { status =>
-              if (status != SubscriptionProcessing)
+            .flatMap {
+              case NewSubscription | SubscriptionRejected | SubscriptionExists =>
```

Comment:
Since we have the new 'SubscriptionProcessing' case and then these other 3 as well. Can we check that we have tests for each, if they werent there already (I cant tell from the diff)